### PR TITLE
Empty vectors before filling them in ChessBoardDetector::processQuads

### PR DESCRIPTION
**Pull request by vrabaud**
It seems the port in https://github.com/opencv/opencv/pull/11703 lost the initialization.

The only breaking tests I have are unfortunately internal and complex.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by vrabaud** [comment]
@MaximSmolskiy , you seem to have fixed a lot lately in that code, does that make sense?

File: modules/calib3d/src/calibinit.cpp [hunk 1]
```diff
@@ -1937,6 +1938,7 @@ bool ChessBoardDetector::processQuads(std::vector<cv::Point2f>& out_corners, int
         if (count > 0 || (-count > (int)out_corners.size()))
         {
             // copy corners to output array
+            out_corners.clear();
```
Comment (asmorkalov) [inline modules/calib3d/src/calibinit.cpp:1941]: There is `out_corners.resize(0);` at the function beginning (line 1886). M.b. it should be replaced with clear() call and the condition may be reduced.

**Review by asmorkalov — CHANGES_REQUESTED** [review]

File: modules/calib3d/src/calibinit.cpp (hunk 1 above)
Comment (vrabaud) [reply modules/calib3d/src/calibinit.cpp:1941]: Right, there is one line 1886 but what if you enter this block (1938 to 1951), fill out_corners and actually fail the condition like 1946?
You then can push_back again in the next loop iteration (from loop at 1899) while `out_corners` has not been cleared.

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**